### PR TITLE
Trigger docs only when something inside the docs folder changed

### PR DESCRIPTION
**Pull request by KasukabeDefenceForce**
### :pencil: Description

**Type:** :roller_coaster: `infrastructure`
I am trying to implement a conditional logic in the docs workflow so that the docs trigger only if there is a change in any of the files in the docs directory.

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by andrewfullard** [comment]
Why are the tests failing in this PR?

**Comment by atharva-2001** [comment]
Can you please include example workflows- conditions when the workflow ran successfully?

**Review by atharva-2001 — CHANGES_REQUESTED** [review]
Thanks, Abhinav, I believe the workflow will now run if there are changes in the docs folder. But someone will still be able to run it manually too using the build-docs label(without skipping), as a test?

**Comment by atharva-2001** [comment]
> Why are the tests failing in this PR?
The regression cache was corrupt(size was in bytes). Deleted the cache and now it works.

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -37,47 +36,63 @@ defaults:
...
         if: github.event_name == 'pull_request_target'
 
+      - name: Check for changes in documentation
+        run: |
+          if git diff origin/master..."$(git rev-parse --abbrev-ref HEAD)" --name-only | cat | grep 'docs' | grep -q .; then
+              num_files=$(git diff --name-only origin/master...HEAD | grep '^docs/' | wc -l)
+              echo "Changes found in documentation files: $num_files"
+              exit 0
+          else
+              echo "No changes found in documentation files"
```
Comment (wkerzendorf) [inline .github/workflows/build-docs.yml:63]:
```suggestion
              echo "No changes found in documentation files - will stop running the pipeline."
```

**Comment by KasukabeDefenceForce** [comment]
> Thanks, Abhinav, I believe the workflow will now run if there are changes in the docs folder. But someone will still be able to run it manually too using the build-docs label(without skipping), as a test?

What should be its behavior when we manually run it? If there are no changes in the docs then it should not do anything or it should run all the steps?

**Comment by andrewfullard** [comment]
With `build-docs` label: should always run the full docs build on PR commits
With force run in actions list: run the full docs build
With docs change: automatically run the full docs build

File: .github/workflows/build-docs.yml [hunk 2]
```diff
@@ -37,47 +36,63 @@ defaults:
...
+          activate-environment: tardis
+          use-mamba: true
+        if: steps.docs_check.outcome == 'success'
 
       - uses: actions/cache@v2
         with:
           path: /usr/share/miniconda3/envs/tardis
           key: conda-linux-64-${{ hashFiles('conda-linux-64.lock') }}-${{ env.CACHE_NUMBER }}
         id: cache-conda
+        if: steps.docs_check.outcome == 'success'
```
[the remark this answers is not in the document]
Comment (KasukabeDefenceForce) [reply .github/workflows/build-docs.yml]: Yes, the redundant step is removed and now it initially checks for changes and exits if there are no changes, showing a green tick.

**Comment by KasukabeDefenceForce** [comment]
> Thanks, Abhinav, I believe the workflow will now run if there are changes in the docs folder. But someone will still be able to run it manually too using the build-docs label(without skipping), as a test?

Yes, we can run it manually, by either manually running the workflow or by adding the build docs label to the PR.

File: .github/workflows/build-docs.yml (hunk 1 above)
Comment (KasukabeDefenceForce) [reply .github/workflows/build-docs.yml]: i have updated the message.

**Comment by atharva-2001** [comment]
There is a merge conflict. Also, can you provide example workflow links?

File: .github/workflows/build-docs.yml [hunk 3]
```diff
@@ -36,42 +35,73 @@ defaults:
...
+              exit 1
+          fi
+        id: docs_check
+        if: steps.trigger_check.outcome != 'success'
+        continue-on-error: true
 
+  build-docs:
     runs-on: ubuntu-latest
+    needs: Check-for-changes
+    if: needs.Check-for-changes.outputs.trigger-check-outcome == 'success' || needs.Check-for-changes.outputs.docs-check-outcome == 'success'
```
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:79]:
```suggestion
    if: needs.check-for-changes.outputs.trigger-check-outcome == 'success' || needs.check-for-changes.outputs.docs-check-outcome == 'success'
```

File: .github/workflows/build-docs.yml [hunk 4]
```diff
@@ -36,42 +35,73 @@ defaults:
     shell: bash -l {0}
 
 jobs:
-  build-docs:
+  Check-for-changes:
```
Comment (atharva-2001) [inline .github/workflows/build-docs.yml:38]:
```suggestion
  check-for-changes:
```

File: .github/workflows/build-docs.yml (hunk 3 above)
Comment (KasukabeDefenceForce) [reply .github/workflows/build-docs.yml]: I have renamed the job using lower case.

File: .github/workflows/build-docs.yml (hunk 4 above)
Comment (KasukabeDefenceForce) [reply .github/workflows/build-docs.yml]: I have renamed the job using lower case.

**Review by andrewfullard — APPROVED** [review]

**Review by atharva-2001 — APPROVED** [review]

**Pull request merged**